examples/pytorch/text-classification/run_classification.py

```diff
@@ -0,0 +1,666 @@
+    # https://huggingface.co/docs/datasets/loading_datasets.html.
+
+    if data_args.remove_splits is not None:
+        for split in data_args.remove_splits.split(","):
+            logger.info(f"removing split {split}")
+            raw_datasets.pop(split)
+
+    if data_args.remove_columns is not None:
+        for split in raw_datasets.keys():
+            for column in data_args.remove_columns.split(","):
+                logger.info(f"removing column {column} from split {split}")
+                raw_datasets[split].remove_columns(column)
+
+    if data_args.label_column_name is not None and data_args.label_column_name != "label":
+        for key in raw_datasets.keys():
+            raw_datasets[key] = raw_datasets[key].rename_column(data_args.label_column_name, "label")
+
+    is_multi_label = False
+    # Trying to have good defaults here, don't hesitate to tweak to your needs.
+    is_regression = raw_datasets["train"].features["label"].dtype in ["float32", "float64"]
+    if is_regression:
+        num_labels = 1
+    else:  # classification
+        if isinstance(raw_datasets["train"]["label"][0], list):  # multi-label classification
+            is_multi_label = True
+            logger.info("Label type is list, doing multi-label classification")
+        # Trying to find the number of labels in a multi-label classification task
+        # We have to deal with common cases that labels appear in the training set but not in the validation/test set.
+        # So we build the label list from the union of labels in train/val/test.
+        label_list = get_label_list(raw_datasets, split="train")
+        for split in ["validation", "test"]:
+            if split in raw_datasets:
+                val_or_test_labels = get_label_list(raw_datasets, split=split)
+                diff = set(val_or_test_labels).difference(set(label_list))
+                if len(diff) > 0:
+                    # add the labels that appear in val/test but not in train, throw a warning
+                    logger.warning(
+                        f"Labels {diff} in {split} set but not in training set, adding them to the label list"
+                    )
+                    label_list += list(diff)
+        # if label is -1, we throw a warning and remove it from the label list
+        for label in label_list:
+            if label == -1:
+                logger.warning("Label -1 found in label list, removing it.")
+                label_list.remove(label)
+
+        label_list.sort()
+        num_labels = len(label_list)
+
+    # Load pretrained model and tokenizer
+    # In distributed training, the .from_pretrained methods guarantee that only one local process can concurrently
+    # download model & vocab.
+    config = AutoConfig.from_pretrained(
+        model_args.config_name if model_args.config_name else model_args.model_name_or_path,
+        num_labels=num_labels,
+        finetuning_task=None,
```

Comment:
Here there is a `problem_type` argument you should use to set the right task (regression, single label or multi label). This will make sure the model fine-tuned/trianed works properly in a pipeline after.

Reply:
I have set "text-regression", "text-classification" and "multi-label-text-classification" for the three finetuning tasks. Is it ok? please would you kindly review~

Reply:
Note that I use "-" instead of "_" to follow the code in other examples such as [https://github.com/huggingface/transformers/blob/129cb6d5239374869beea0e1317641426e8df572/examples/pytorch/image-classification/run_image_classification_no_trainer.py#L300](image-classification)

Reply:
I think for the fine-tuning task those are not the labels expected by the Hub. It should be `text-classification` in all cases.

Reply:
yes,  I have fixed it.